test/utils/introspection/ERC165Checker.test.js

```diff
@@ -50,23 +50,49 @@ describe('ERC165Checker', function () {
     });
 
     it('does not support ERC165', async function () {
-      expect(await this.mock.$supportsERC165(this.target)).to.be.false;
+      await expect(this.mock.$supportsERC165(this.target)).to.eventually.be.false;
     });
 
     it('does not support mock interface via supportsInterface', async function () {
-      expect(await this.mock.$supportsInterface(this.target, DUMMY_ID)).to.be.false;
+      await expect(this.mock.$supportsInterface(this.target, DUMMY_ID)).to.eventually.be.false;
     });
 
     it('does not support mock interface via supportsAllInterfaces', async function () {
-      expect(await this.mock.$supportsAllInterfaces(this.target, [DUMMY_ID])).to.be.false;
+      await expect(this.mock.$supportsAllInterfaces(this.target, [DUMMY_ID])).to.eventually.be.false;
     });
 
     it('does not support mock interface via getSupportedInterfaces', async function () {
-      expect(await this.mock.$getSupportedInterfaces(this.target, [DUMMY_ID])).to.deep.equal([false]);
+      await expect(this.mock.$getSupportedInterfaces(this.target, [DUMMY_ID])).to.eventually.deep.equal([false]);
     });
 
     it('does not support mock interface via supportsERC165InterfaceUnchecked', async function () {
-      expect(await this.mock.$supportsERC165InterfaceUnchecked(this.target, DUMMY_ID)).to.be.true;
+      await expect(this.mock.$supportsERC165InterfaceUnchecked(this.target, DUMMY_ID)).to.eventually.be.true;
+    });
+  });
+
+  describe('ERC165 revert on invalid interface', function () {
+    beforeEach(async function () {
+      this.target = await ethers.deployContract('ERC165RevertInvalid');
+    });
+
+    it('does not support ERC165', async function () {
+      await expect(this.mock.$supportsERC165(this.target)).to.eventually.be.false;
+    });
+
+    it('does not support mock interface via supportsInterface', async function () {
+      await expect(this.mock.$supportsInterface(this.target, DUMMY_ID)).to.eventually.be.false;
+    });
+
+    it('does not support mock interface via supportsAllInterfaces', async function () {
+      await expect(this.mock.$supportsAllInterfaces(this.target, [DUMMY_ID])).to.eventually.be.false;
+    });
+
+    it('does not support mock interface via getSupportedInterfaces', async function () {
+      await expect(this.mock.$getSupportedInterfaces(this.target, [DUMMY_ID])).to.eventually.deep.equal([false]);
+    });
+
+    it('does not support mock interface via supportsERC165InterfaceUnchecked', async function () {
+      await expect(this.mock.$supportsERC165InterfaceUnchecked(this.target, DUMMY_ID)).to.eventually.be.false;
```

Comment:
Here we have no check that return true. We should check that some there is some `supportsERC165InterfaceUnchecked` that doesn't trigger the revert that returns true.

Reply:
fixed in [ed19036](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/5880/commits/ed190367c5c01ec0651d0ad421e7bc2fba338139)